apps/api/src/app/blueprint/blueprint.controller.ts

```diff
@@ -1,45 +1,44 @@
-import { ClassSerializerInterceptor, Controller, Get, Param, Post, UseGuards, UseInterceptors } from '@nestjs/common';
+import { ClassSerializerInterceptor, Controller, Get, Param, Post, UseInterceptors } from '@nestjs/common';
 
 import { IJwtPayload } from '@novu/shared';
+import { NotificationTemplateEntity } from '@novu/dal';
 
 import { UserSession } from '../shared/framework/user.decorator';
-import { NotificationTemplateResponse } from '../notification-template/dto/notification-template-response.dto';
-import {
-  GetBlueprintNotificationTemplate,
-  GetBlueprintNotificationTemplateCommand,
-} from './usecases/get-blueprint-notification-template';
-import { CreateBlueprintNotificationTemplate } from './usecases/create-blueprint-notification-template';
+import { GroupedBlueprintResponse } from './dto/grouped-blueprint.response.dto';
+import { GetBlueprint, GetBlueprintCommand } from './usecases/get-blueprint';
+import { CreateBlueprintCommand, CreateBlueprint } from './usecases/create-blueprint';
+import { GetGroupedBlueprints } from './usecases/get-blueprints';
 
 @Controller('/blueprints')
 @UseInterceptors(ClassSerializerInterceptor)
 export class BlueprintController {
   constructor(
-    private getBlueprintNotificationTemplate: GetBlueprintNotificationTemplate,
-    private createBlueprintNotificationTemplate: CreateBlueprintNotificationTemplate
+    private getBlueprintUsecase: GetBlueprint,
+    private getGroupedBlueprintsUsecase: GetGroupedBlueprints,
+    private createBlueprintUsecase: CreateBlueprint
   ) {}
 
   @Get('/:templateId')
-  getNotificationTemplateBlueprintById(
-    @UserSession() user: IJwtPayload,
-    @Param('templateId') templateId: string
-  ): Promise<NotificationTemplateResponse> {
-    return this.getBlueprintNotificationTemplate.execute(
-      GetBlueprintNotificationTemplateCommand.create({
-        environmentId: user.environmentId,
-        organizationId: user.organizationId,
-        userId: user._id,
+  getBlueprintById(@Param('templateId') templateId: string): Promise<NotificationTemplateEntity> {
```

Comment:
Can we return a DTO like `NotificationStepDto` or similar instead of returning the DAL entity? It doesn't feel right we have to have a dependency to DAL in this layer.
Same for the method `createNotificationTemplateFromBlueprintById`.

Reply:
Yeah definitely I will create one, tho createNotificationTemplateFromBlueprintById will be deleted in this cycle so I won't add it there.